src/compute/src/sink/persist_sink.rs

```diff
@@ -40,68 +49,209 @@ where
     fn render_continuous_sink(
         &self,
         compute_state: &mut ComputeState,
-        _sink: &ComputeSinkDesc<CollectionMetadata>,
+        sink: &ComputeSinkDesc<CollectionMetadata>,
         sink_id: GlobalId,
         sinked_collection: Collection<G, Row, Diff>,
         err_collection: Collection<G, DataflowError, Diff>,
     ) -> Option<Rc<dyn Any>>
     where
         G: Scope<Timestamp = Timestamp>,
     {
+        let desired_collection = sinked_collection.map(Ok).concat(&err_collection.map(Err));
+
         persist_sink(
             sink_id,
             &self.storage_metadata,
-            sinked_collection,
-            err_collection,
+            desired_collection,
+            sink.as_of.frontier.clone(),
             compute_state,
-            false,
         )
     }
 }
 
 pub(crate) fn persist_sink<G>(
```

Comment:
I thought about that. We should be able to use the same implementation but we have to create a `persist_source` that doesn't read the snapshot and filters out everything beyond the resume frontier. For a source, `desired` will only contain new updates from the source, not the entire "history".

This will also mean that for sources the `persist_sink` will only be self-correcting from the resume timestamp forward. What do you think?

Reply:
I think that sounds sensible, but I'm also very uninformed about storage/source details, so definitely don't trust me on this!